### PR TITLE
feat(wallet): Show Assets with 0 Balances in Panel

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 

1) Now shows assets with `0` balances in the `Assets Panel` view
2) Now has a `Add asset` button in the `Assets Panel` view

Resolves <https://github.com/brave/brave-browser/issues/20955>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/153249299-e4782146-2190-4976-9e69-2bb80738bde9.mov

### CODE REVIEW

File: components/brave_wallet_ui/common/hooks/assets.ts [hunk 1]
```diff
@@ -62,12 +62,7 @@ export default function useAssets (
       return []
     }
 
-    const nonZeroBalanceAssets = userVisibleTokensInfo.filter(token => {
-      const balance = getBalance(selectedAccount, token)
-      return balance !== '' && parseInt(balance) > 0
-    })
-
-    return nonZeroBalanceAssets.sort(function (a, b) {
+    return userVisibleTokensInfo.sort(function (a, b) {
```
Comment (onyb) [inline components/brave_wallet_ui/common/hooks/assets.ts:65]: cc: @muliswilliam to check if this has any consequence for your other PR. Recall that we wanted to use `panelUserAssetList` for `sendAssetOptions`. I think filtering out all the zero-balance assets never made sense, so let's do the same for send assets list too.

**Review by onyb — APPROVED** [review]
LGTM!

File: components/brave_wallet_ui/common/hooks/assets.test.ts [hunk 2]
```diff
@@ -31,8 +31,6 @@ const mockVisibleList = [
   AccountAssetOptions[1]
 ]
 
-const expectedResult = [AccountAssetOptions[0]]
```
Comment (muliswilliam) [inline components/brave_wallet_ui/common/hooks/assets.test.ts:34]: Can we move this to the specific test where it is used?

File: components/brave_wallet_ui/common/hooks/assets.ts (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/common/hooks/assets.ts:65]: @onyb This is okay. As per our discussion, we need to agree on how to handle positive balances.

**Review by muliswilliam — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/common/hooks/assets.test.ts (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/assets.test.ts:34]: This was removed, no longer used.

**Review by muliswilliam — APPROVED** [review]

**Pull request merged**